Set/Set/SetGame.swift

```diff
@@ -84,17 +127,55 @@ final class SetGame {
         }
     }
 
+    private func updateMatchingState(for cardIndex: Int) {
+        guard self.chosenCardsIndices.count > 2 else {
+            return
+        }
+
+        if matchCards() {
+            self.matchedCardsIndices.append(contentsOf: self.chosenCardsIndices)
+            self.score += 3
```

Comment:
I would prefer to have a separate method for scoring. It can be named similarly for this method